free-1.md

```diff
@@ -1,49 +1,99 @@
+
+_How did you come up with it?_
+
+I came up with my idea for this project while looking at the DSL that played the Batman theme song from an Excel spreadsheet! 
 
 ### Domain
-_Describe the project's domain in five words._
+Translating chord names -> sheet music 
 
 
 ### Interface (syntax)
-_How might the user interact with the language? What does programming look 
-like? Why is this the right way to interact with the problem domain?_ 
+_How might the user interact with the language?_
+
+The language will provide a text input box that allows someone to type in the name of a chord. The user will then select the clef that the chord is written in. The language will then display a generated image of the chord on a musical staff, and the user can save the chord and move onto translating another chord, most likely in the order the chords appear in a song. 
+
+_What does programming look like? Why is this the right way to interact with the problem domain?_ 
+
+The user simply has to type in the name of the chord, and then use checkboxes to decide what clef the chord is in (and other features I'm not thinking about now). This is the right way to interact with the problem domain because when pianists look up the chords for a song, they want to be able to play the music right away, and I want my program to be as hassle-free as possible. Also, typing in the name of the chord requires no understanding of music theory, which I do not expect for my users to have. 
 
 
 ### Operation (semantics)
-_What might happen when a program runs? How does a program interact with the
-user? What kinds of errors might occur, and how might they be communicated to
-the user?_
+_What might happen when a program runs?_
+
+The program needs to break up the input into several features: the clef, the key signature or root of the chord, the type of chord it is (Major, minor, diminished  7th, etc.), and the inversion of the chord. After receiving the input, the program would then parse the three pieces of information, start with the clef and root of the chord, and build up interval by interval according to the type of chord it is. For example, the 2nd note in a major chord is a major 3rd interval above the root, whereas the 2nd note in a minor chord is a minor 3rd above the root. Then, the program would look at what inversion was specified (and if there was no inversion, assume a root position of the chord), and rearrange the order of the notes in the chord accordingly. 
+
+_How does a program interact with the user?_
+
+The program will prompt the user for the name of the chord that he/she wants to translate, offering 4 text boxes for the user to specify: the clef of the staff, the key signature or root of the chord, the type of chord it is (Major, minor, diminished  7th, etc.), and the inversion of the chord.
+
```

Comment:
I think printing the notes on the musical staff, rather than just the names of the notes, is a great idea.  I have some questions about how the user would interact with this program.  Could they only enter in one chord at a time, and then generate the image?  In that case, if a pianist wanted to convert an entire song from guitar tabs, would they convert each chord individually with your DSL, and then transcribe it onto staff paper?  (Or is this outside of your domain?)  Also, would a user be able to indicate other relevant things, like the duration of the chord or if there are any rests?